UserDocs.md

```diff
@@ -144,54 +106,72 @@ In order to successfully download a package from a registry, that package must o
 	t4.list_packages()  # to see local packages
-	t4.list_packages("s3://bucket")  # to see remote packages
-
-<!-- ![](img/helium-api.png) -->
+	t4.list_packages("s3://name-of-your-t4-bucket")  # to see remote packages
 
 
-### Managing individual files
+### Working with buckets
 
-T4 provides a lower-level API for managing both local and remote files using `t4.copy`:
-
-```python
-t4.copy("file:///my-frame.csv", "s3://bucket-name/my-frame.csv")  # copy local to local
-t4.copy("s3:///my-frame.csv", "s3://bucket-name/my-frame.csv")  # copy remote to remote
-t4.copy("s3://my-frame.csv", "file:///bucket-name/my-frame.csv")  # copy remote to local
-```
+You can manage individual files using the `Bucket` interface. A `Bucket` is just a wrapper on your registry:
 
-T4 supports files in any format, and up to 5 TB in size.
+    import t4
+    b = t4.Bucket("s3://name-of-your-t4-bucket")
 
-To delete a remote file, use the `delete` command:
+The `Bucket` object supports the same getters that `Package` supports: <!-- TODO: potentially with the twise that set is now push -->
 
-```python
-he.delete("s3://bucket-name/my-frame.csv")
-```
+    # download a directory or a file with fetch
+    b.fetch("path/to/directory", "path/to/local")
+    b.fetch("path/to/file", "path/to/local")
+    
+    # load an object into memory with deserialize
+    obj = b.deserialize("path/to/file")
+    obj = b("path/to/file")
+    
+    # get metadata
+    meta = b.get_meta("path/to/file")
+    
+    # delete a file
+    b.delete("path/to/file")
+    
+    # and so on...
 
 
 ### Working with memory
 
-You can save a Python object to a T4 file using the `put` command:
+You can read data right out of a `Package` or `Bucket` using `deserialize`. 
 
-```python
-import t4
-t4.put(df, "bucket-name/my-frame.parquet")
-```
+    p = t4.Package.browse("username/packagename")
+    b = t4.Bucket("s3://store")
+    
+    obj = b.deserialize("bucket-name/my-frame.parquet")
+    obj = b("bucket-name/my-frame.parquet")  # shortcut
+    
+    obj = p["my/package"].deserialize()
+    obj = p["my/package"]()  # shortcut
 
-The above code writes `df` at the top level of `bucket-name`.
+On the flip side, you can serialize objects to T4 using `Bucket.put`.
+
+    b.put(obj, "bucket-name/my-frame.parquet")
 
 If you `put` to a folder that doesn't exist yet, `t4` will create that folder. If you overwrite an object, and bucket versioning is enabled, the overwritten object is retained as an older version of the same path. 
 
 T4 transparently serializes and de-serializes select Python
 objects. In the above example, `df` is automatically stored as an Apache Parquet file.
 
-To read `df` out of S3 and into local memory, use `get`:
 
-```python
-df, meta = he.get("bucket-name/my-frame.parquet")
-```
+### Moving data not on T4
+We encourage a push/pull architecture based on having data packages ultimately live on T4. But you may also use the t4 CLI to handle moving data anywhere on your local file system.
```

Comment:
"We encourage a build-push-pull development cycle against a central registry in S3, or on a remote disk or NAS." is clearer. There's no reason to prefer S3. The real meat is "not on my box" and "central store". "Architecture" is too strong of a word here imho.